### PR TITLE
[1.12.2] Migrate SeedEntry to LootTable

### PR DESCRIPTION
**Pull request by 3TUSK**
Given the fact that, in 1.14 snapshots, all block drops will use loot table, it would be a good start to deprecate the old `SeedEntry` system and migrate to LootTable (which seems to be the plan according to the comments left in related classes) ahead of time right now. Hence here is a PR. 

Summary:
  - `getGrassSeed` now draws `ItemStack` from LootTable `forge:grass_seeds`.
  - Two `MinecraftForge.addGrassSeed` are deprecated.
  - Backward compatibility is included (because it is targeting 1.12.2) by retrofitting the internal `List<SeedEntry>` to a "fake" `LootPool` that will be injected at runtime.

If a PR targeting `1.13` branch is preferred, feel free to close this, and I will make a new one.

However, there are still some tricky issues need to be discussed (and properly addressed if necessary):
  - `getGrassSeed` only return a single `ItemStack` instance; while `generateLootForPools` returns a `List<ItemStack>` which has varied size depending on the actual LootTable. For now, my proposed solution is to shuffle the `List<ItemStack>` again using the supplied `Random` and take the `ItemStack` at index 0.
  - There is no way to reliably pass fortune level to `LootContext` for backward compatibility sake. For now, I re-use the `luck` field which supposed to be used for luck value or Looting enchantment level, but I have a strong feel that doing so would instantly kill this PR before everything. 
    - Regarding my choice of re-use the `luck` field in `LootContext`: I chose this approach so that there is no need to patch any vanilla classes.
    - However, to properly get the fortune level, one needs to pass the harvester's `EntityPlayer` reference to the `LootContext`, which isn't available in `getGrassSeed` - rather, it is available (?) in the context of `Block.getDrops` (specifically - the `ThreadLocal<EntityPlayer>` that holds the harvester instance - thus the question mark after "available"). Thus this leads the question of "whether migrating `SeedEntry` to LootTable is viable or not".

### CODE REVIEW

File: src/main/java/net/minecraftforge/common/MinecraftForge.java [hunk 1]
```diff
@@ -67,12 +67,21 @@
...
+     *             LootTableLoadEvent}.
+     *
+     * TODO: Remove both methods before 1.14 when all block drops use loot table system
      */
+    @Deprecated
     public static void addGrassSeed(@Nonnull ItemStack seed, int weight)
     {
         addGrassSeed(new SeedEntry(seed, weight));
     }
+    /** @deprecated see {@link #addGrassSeed(ItemStack, int)}*/ @Deprecated
```
Comment (tterrag1098) [inline src/main/java/net/minecraftforge/common/MinecraftForge.java]: This isn't a patch, use some newlines here.

File: src/main/resources/assets/forge/loot_tables/grass_seed.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "pools": []
```
Comment (tterrag1098) [inline src/main/resources/assets/forge/loot_tables/grass_seed.json]: Couldn't we put the current default value in `seedList` here, so forge isn't relying on the deprecated impl?

**Review by tterrag1098 — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/forge/loot_tables/grass_seed.json (hunk 2 above)
Comment (3TUSK) [reply src/main/resources/assets/forge/loot_tables/grass_seed.json]: Will see how feasible is that - given that I need to cover the fortune enchantment support as well.

File: src/main/java/net/minecraftforge/common/ForgeHooks.java [hunk 3]
```diff
@@ -180,19 +185,62 @@ public ItemStack getStack(Random rand, int fortune)
...
+     */
+    static final ResourceLocation GRASS_SEED_TABLE = new ResourceLocation(ForgeVersion.MOD_ID, "grass_seed");
+
     @Nonnull
     public static ItemStack getGrassSeed(Random rand, int fortune)
     {
-        if (seedList.size() == 0)
+        // Passes fortune level as "luck" to the LootContext so that it can be used
+        // TODO Properly introduce EntityPlayer as new param of getGrassSeed
+        LootContext.Builder contextBuilder = new LootContext.Builder(FMLCommonHandler.instance().getMinecraftServerInstance().getWorld(0)).withLuck(fortune);
```
Comment (mezz) [inline src/main/java/net/minecraftforge/common/ForgeHooks.java:233]: Please store `FMLCommonHandler.instance().getMinecraftServerInstance().getWorld(0)` to a local variable, it is used two times in this method.

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by LexManos** [comment]
Going to leave this as a no for 1.12.x as 1.14 is the new targeted branch and there hasnt been any activity on this.

**Pull request closed without merging**